gaudi/gaudi_spi_fw_upgrade_build.yaml

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+---
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: gaudi-spi-fw-upgrade
+  namespace: gaudi-validation
+---
+kind: BuildConfig
+apiVersion: build.openshift.io/v1
+metadata:
+  name: gaudi-spi-fw-upgrade
+  namespace: gaudi-validation
+spec:
+  output:
+    to:
+      kind: ImageStreamTag
+      name: gaudi-spi-fw-upgrade:1.18.0-524
+  strategy:
+    type: Docker
+  source:
+    type: Dockerfile
+    dockerfile: >+
+      ARG
+      BUILDER=vault.habana.ai/gaudi-docker/1.18.0/rhel9.4/habanalabs/pytorch-installer-2.4.0:1.18.0-524
+
+      FROM ${BUILDER} AS builder
+
+      RUN echo "[habanalabs]" > /etc/yum.repos.d/habanalabs.repo && \
+        echo "name=Habana RH9 Linux repo" >> /etc/yum.repos.d/habanalabs.repo && \

```

Comment:
Can we use 

```
cat <<EOF > /etc/yum.repos.d/habanalabs.repo
name=Habana RH9 Linux repo
baseurl=https://vault.habana.ai/artifactory/rhel/9/9.4
gpgkey=https://vault.habana.ai/artifactory/api/v2/repositories/rhel/keyPairs/primary/public
gpgcheck=1
EOF
```

Reply:
That is directly referred from Habana's dockerfile at -  https://github.com/HabanaAI/Setup_and_Install/blob/5ec0d0c9be8faf9120bfc72d27bdff1eb5abec4f/dockerfiles/base/Dockerfile.rhel9.4#L101. Will update if still needed.

Reply:
I think the cat << EOF is the better. :-)

Reply:
@chaitanya1731 any comments on it? :-)

Reply:
according to this documentation we tried the recommended way - https://www.docker.com/blog/introduction-to-heredocs-in-dockerfiles/ but we can still try to optimize it in tech preview release.